### PR TITLE
Error wraping and reporting

### PR DESCRIPTION
**Pull request by matei-p**
Error wrapping and reporting (for now in breakpoint handling code and as a poc in TransferESDTNFTExecute).

Note that in order to increase the testability of Elrond API (elrondei.go), a code split was made for transferESDTNFTExecute(). Getting the host from the context pointer should not be in the same place with the actual API function logic. For the same reason, in the future, further separation will be done between the argument parsing logic (using MemLoads) and the actual API logic.

### CODE REVIEW

File: mock/contracts/esdtSimpleSC.go [hunk 1]
```diff
@@ -40,3 +41,40 @@ func ExecESDTTransferAndCallParentMock(instanceMock *mock.InstanceMock, config i
...
+		input.Arguments = [][]byte{
+			test.ESDTTestTokenName,
+			big.NewInt(int64(testConfig.ESDTTokensToTransfer)).Bytes(),
+			arguments[2],
+		}
+		input.RecipientAddr = arguments[0]
+
+		elrondapi.TransferESDTNFTExecute(
+			host,
+			arguments[0],
```
Comment (camilbancioiu) [inline mock/contracts/esdtSimpleSC.go:71]: For readability.

```suggestion
			input.RecipientAddr,
```

File: mock/contracts/esdtSimpleSC.go [hunk 2]
```diff
@@ -40,3 +41,40 @@ func ExecESDTTransferAndCallParentMock(instanceMock *mock.InstanceMock, config i
...
+			test.ESDTTestTokenName,
+			testConfig.ESDTTokensToTransfer,
+			arguments[1],
+			[][]byte{arguments[2]},
```
Comment (camilbancioiu) [inline mock/contracts/esdtSimpleSC.go:75]: Should be defined as named variables, then passed as parameters, for readability.

Comment (matei-p) [reply mock/contracts/esdtSimpleSC.go:75]: they are passed as parameters via `host.Runtime().Arguments()`

**Review by dorin-iancu — DISMISSED** [review]

File: arwen/helpers.go [hunk 3]
```diff
@@ -177,14 +177,18 @@ func GetStorageContext(vmHostPtr unsafe.Pointer) StorageContext {
 
 // WithFault returns true if the error is not nil, and uses the remaining gas if the execution has failed
 func WithFault(err error, vmHostPtr unsafe.Pointer, failExecution bool) bool {
+	runtime := GetVMHost(vmHostPtr)
+	return WithFaultAndHost(runtime, err, failExecution)
+}
+
+func WithFaultAndHost(host VMHost, err error, failExecution bool) bool {
```
Comment (sasurobert) [inline arwen/helpers.go:184]: add comment to public funciton

File: arwen/elrondapi/elrondei-wrapper.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	offsetArgumentsData := offsetArgumentsLength + int32(len(argumentsLengthsAsBytes))
+	crtArgOffset := offsetArgumentsData
+	for idxArg, arg := range args {
+		argumentsData := make([]byte, argumentsLengths[idxArg])
+		copy(argumentsData, arg)
+		runtime.MemStore(crtArgOffset, argumentsData)
+		crtArgOffset += int32(argumentsLengths[idxArg])
+	}
+
+	return TransferESDTNFTExecuteWithHost(host,
```
Comment (sasurobert) [inline arwen/elrondapi/elrondei-wrapper.go:66]: add one line after "("

**Review by sasurobert — CHANGES_REQUESTED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Review by dorin-iancu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**